control_panel_api/views.py

```diff
@@ -22,28 +28,46 @@
 from control_panel_api.serializers import (
     AppS3BucketSerializer,
     AppSerializer,
-    UserAppSerializer,
     GroupSerializer,
     S3BucketSerializer,
+    UserAppSerializer,
     UserS3BucketSerializer,
     UserSerializer,
 )
 
+logger = logging.getLogger(__name__)
+
 
 class UserViewSet(viewsets.ModelViewSet):
     queryset = User.objects.all()
     serializer_class = UserSerializer
     filter_backends = (UserFilter,)
     permission_classes = (UserPermissions,)
 
+    @transaction.atomic
     def perform_create(self, serializer):
         instance = serializer.save()
-        instance.aws_create_role()
-        instance.helm_create()
 
+        try:
+            instance.aws_create_role()
+            instance.helm_create()
+        except ClientError as e:
+            logger.error(e)
+            raise AWSException
+        except CalledProcessError as e:
+            logger.error(e)
+            instance.aws_delete_role()
```

Comment:
I see that you're trying to manually "roll back" the IAM role creation here.
Only worry of doing this is that if the role was created by the lambda function and `helm upgrade` fails for whatever reason we end up deleting someone perfectly working IAM role.

I would be on the safe side and leave this out, we can always change this later if we *really need* to.

Reply:
Ok will delete, maybe trying to be too clever.